### ISSUE TITLE
How was the checkpoint superb/hubert-large-superb-er trained?

### ISSUE THREAD

**Issue by cnut1648** [s3prl/s3prl#302]
Hi, 
First of all, thank you for sharing the checkpoint!

I'm trying to reproduce your checkpoint `superb/hubert-large-superb-er` [here](https://huggingface.co/superb/hubert-large-superb-er). From the documentation, it seems that this model is fine tuning the [`facebook/hubert-large-ll60k`](https://huggingface.co/facebook/hubert-large-ll60k) on IEMOCAP. 
I have been trying to reproduce the results on IEMOCAP using the `hubert-large-ll60k` checkpoint with AdamW learning rate = `1e-5, 1e-4, 1e-3, 5e-3, 5e-4` and batch size = `16, 32`. But the best from all the combinations are 57 test Acc. 
I have read your [emotion downstream folder](https://github.com/s3prl/s3prl/tree/master/s3prl/downstream/emotion) but it seems that this is for the `DeepNet` model not HuBERT. I am wondering if it's possible to share with us the model hyperparameters.

Thank you! Have a nice weekend.

**Comment by leo19941227** [comment]
Hey,

First of all, if you trace the emotion folder and its `config.yaml` more closely, you will find that the default config is actually using the `UtteranceLevel` model in `downstream/model.py`. So the downstream model is simply a linear layer and a mean pooling layer. The major model parameters come from the pre-trained hubert-large-ll60k, which you can specify as `-u hubert_large_ll60k`. So the final model is actually the concatenation of hubert-large-ll60k and a `UtteranceLevel` small probing model.

Could you please share how you use our emotion downstream folder in more details?
Like what's you command? Did you do 5-fold cross validation?

**Comment by cnut1648** [comment]
Thanks for the reply! I guess I was not reading very carefully.

More details about our tuning:

First of all, I am using pre-saved audios from IEMOCAP, but the people pre-saving the audios told me that they were generated from 16k sampling rate using `librosa.load`, and they are almost certain that the data is not corrupted. We use 5-fold cross validation, similar to [your setup](https://github.com/s3prl/s3prl/blob/master/s3prl/downstream/docs/superb.md#er-emotion-recognition), i.e. train on 4 sessions and test on the rest. For example, fold 1 trains on session 1,2,3,4 and test on session 5. Also, I noticed that in your [preprocessing](https://github.com/s3prl/s3prl/blob/master/s3prl/downstream/emotion/IEMOCAP_preprocess.py), you made excited as happiness as well, so we followed this as well.

Second, for the model, we are actually using Huggingface API since that aligns with our pipeline the best. More specially, we are following the documentation in [here](https://huggingface.co/superb/hubert-large-superb-er), but with upstream librespeech pretrained checkpoints. Example would be
```python
feature_extractor = Wav2Vec2FeatureExtractor.from_pretrained("superb/hubert-large-superb-er")
model = HubertForSequenceClassification.from_pretrained("facebook/hubert-large-ll60k",num_labels=num_classes)
sample_rate = 16_000
criterion = nn.CrossEntropyLoss()

for batch in dataloader:
    labels = batch.pop("labels")
    audios = batch.pop("audios")
    inputs = feature_extractor(
            audios, sampling_rate=sample_rate, return_tensors="pt",
            padding=True
        ).to(model.device)
    logits = model(**inputs).logits
    loss = criterion(logits, labels)
```
We just use the Wav2Vec2FeatureExtractor that you shared in [here](https://huggingface.co/superb/hubert-large-superb-er) since when I stepped into the extractor's `__call__` method it seems that it is simply padding the audios, so we just used yours in the hope that there was no performance issues or dataset leaking (please correct us if we were wrong).

Do you think what we are doing is correct? Now that I know you are not fine tuning using Huggingface API, I guess the difference might be due to some nuisances that huggingface model doesn't support? I will try to run your commands as well. Just to make sure, would you mind sharing with us the command you run for the fine tuning? I will be using precisely that.

Thank you again!!

**Comment by leo19941227** [comment]
Hey,

The command we used is simply the same as the document:
https://github.com/s3prl/s3prl/blob/master/s3prl/downstream/docs/superb.md#er-emotion-recognition

Let me know what details are missing there. Thanks!

**Comment by bagustris** [comment]
Hey @cnut1648,

Regarding fine-tuning, I think fine-tuning is used only in `validation` set since no `-f` argument is given in the command in the document from @leo19941227  above. It just simply extracts acoustic features using SSL from the upstream model (e.g., hubert) and inputs them into the er classifier (CNNSelfAttention) in the training, and then adjusts the weights (hidden states) in the validation for generating (ckpt) model. isn't it?

> Do you think what we are doing is correct? Now that I know you are not fine tuning using Huggingface API, I guess the difference might be due to some nuisances that huggingface model doesn't support? I will try to run your commands as well. Just to make sure, would you mind sharing with us the command you run for the fine tuning? I will be using precisely that.

**Comment by leo19941227** [comment]
One simple correction, we use the a simple mean pooling and the linear layer as the downstream model, instead of the CNN model you mentioned, as shown in the default config.yaml
